conformance_tests/core/test_image/src/test_image_swizzle.cpp

```diff
@@ -22,151 +23,140 @@ class zeCommandListAppendImageCopyWithSwizzleTests : public ::testing::Test {
   image_descriptor_source.format.layout = ZE_IMAGE_FORMAT_LAYOUT_8_8_8_8;
   image_descriptor_source.format.x = ZE_IMAGE_FORMAT_SWIZZLE_R;
   image_descriptor_source.format.y = ZE_IMAGE_FORMAT_SWIZZLE_G;
   image_descriptor_source.format.z = ZE_IMAGE_FORMAT_SWIZZLE_B;
   image_descriptor_source.format.w = ZE_IMAGE_FORMAT_SWIZZLE_A;
-  image_descriptor_source.height = test.image_height;
-  image_descriptor_source.width = test.image_width;
-  image_descriptor_source.depth = 1;
+  image_descriptor_source.width = image_dims.width;
+  image_descriptor_source.height = image_dims.height;
+  image_descriptor_source.depth = image_dims.depth;
+  image_descriptor_source.arraylevels = array_levels;
   image_descriptor_source.format.type = ZE_IMAGE_FORMAT_TYPE_UINT;
-  test.img_in = lzt::create_ze_image(image_descriptor_source);
+  img_in = lzt::create_ze_image(image_descriptor_source);
 
   ze_image_desc_t image_descriptor_dest = {};
   image_descriptor_dest.stype = ZE_STRUCTURE_TYPE_IMAGE_DESC;
-  image_descriptor_dest.type = ZE_IMAGE_TYPE_2D;
+  image_descriptor_dest.type = image_type;
   image_descriptor_dest.flags =
       (ZE_IMAGE_FLAG_KERNEL_WRITE | ZE_IMAGE_FLAG_BIAS_UNCACHED);
   image_descriptor_dest.format.layout = ZE_IMAGE_FORMAT_LAYOUT_8_8_8_8;
   image_descriptor_dest.format.x = ZE_IMAGE_FORMAT_SWIZZLE_A;
   image_descriptor_dest.format.y = ZE_IMAGE_FORMAT_SWIZZLE_B;
   image_descriptor_dest.format.z = ZE_IMAGE_FORMAT_SWIZZLE_G;
   image_descriptor_dest.format.w = ZE_IMAGE_FORMAT_SWIZZLE_R;
-  image_descriptor_dest.height = test.image_height;
-  image_descriptor_dest.width = test.image_width;
-  image_descriptor_dest.depth = 1;
+  image_descriptor_dest.width = image_dims.width;
+  image_descriptor_dest.height = image_dims.height;
+  image_descriptor_dest.depth = image_dims.depth;
+  image_descriptor_dest.arraylevels = array_levels;
   image_descriptor_dest.format.type = ZE_IMAGE_FORMAT_TYPE_UINT;
-  test.img_out = lzt::create_ze_image(image_descriptor_dest);
-
-  uint32_t *inbuff =
-      (uint32_t *)lzt::allocate_host_memory(test.image_size * sizeof(uint32_t));
-  uint32_t *outbuff =
-      (uint32_t *)lzt::allocate_host_memory(test.image_size * sizeof(uint32_t));
-
-  for (int i = 0; i < test.image_size; i++) {
-    inbuff[i] = 0x12345678;
-  }
-
-  test.run_test(inbuff, outbuff, "image_swizzle_test", is_immediate);
+  img_out = lzt::create_ze_image(image_descriptor_dest);
+}
 
-  for (int i = 0; i < test.image_size; i++) {
-    EXPECT_EQ(outbuff[i], 0x78563412); // After swizzle from RGBA to AGBR the
-                                       // data format will be reversed from 12
-                                       // 34 56 78 -> 78 56 34 12
+void RunGivenDeviceImageAndHostImagesWithDifferentSwizzleWhenLaunchingCopyFromKernel(
+    zeCommandListAppendImageCopyWithSwizzleTests &test, bool is_immediate) {
+  for (auto image_type : test.tested_image_types) {
+    test.run_test(image_type, is_immediate);
```

Comment:
The conformance test count should include unsupported scenarios.
Skip if capabilities are not there. Throw 'not implemented' if test lacks implementation and fail in any other scenario.